Libraries/LLM/Evaluate.swift

```diff
@@ -174,14 +173,32 @@ public enum GenerateDisposition {
 ///   - parameters: generation parameters
 ///   - model: model to evaluate
 ///   - tokenizer: tokenizer to convert tokens back into strings and recognizer special tokens
+///   - configuration: the model configuration
 ///   - didGenerate: visitor for the tokens as they are generated
 public func generate(
     promptTokens: [Int], parameters: GenerateParameters, model: LLMModel, tokenizer: Tokenizer,
+    configuration: ModelConfiguration,
```

Comment:
I'm not certain about the way this model configuration gets passed around and what the intention for it is. Somehow we don't have a need for it in Python, so I'm wondering why do you need it here? What should go in it vs in the tokenizer/model directly?

From what I can tell it's more like the default arguments for a given model (eos token / prompt). The prompt gets handled outside this function. So is here it's just for the additional eos tokens?

Reply:
yeah, just for the additional eos tokens.  I can switch that to just pass in additional eos tokens (optional).  If we need more parameters in the future I can rethink that slightly.